File: 1424-diagonal-traverse-ii/1424-diagonal-traverse-ii.py
class Solution:
    def findDiagonalOrder(self, nums: List[List[int]]) -> List[int]:
        result = []
        groups = {}
        
        for row in range(len(nums)-1, -1, -1):
            for column in range(len(nums[row])-1, -1, -1):
                diagonal = row + column
                if diagonal in groups:
                    groups[diagonal].append(nums[row][column])
                else:
                    groups[diagonal] = [nums[row][column]]
        
        curr = 0
        while curr in groups:
            result.extend(groups[curr])
            curr += 1
        
        return result

File: 1424-diagonal-traverse-ii/1424-diagonal-traverse-ii.js
/**
 * @param {number[][]} nums
 * @return {number[]}
 */
var findDiagonalOrder = function(nums) {
    let result = [];
    let groups = {};
    
    for (let row = nums.length - 1; row > -1; row--){
        for (let column = nums[row].length - 1; column > -1; column--){
            let diagonal = row + column;
            if (diagonal in groups){
                groups[diagonal].push(nums[row][column]);
            }
            else {
                groups[diagonal] = [nums[row][column]];
            }
        }
    }
    
    let curr = 0;
    
    while (curr in groups){
        result.push(...groups[curr]);
        curr += 1;
    }
    
    return result;
};